packages/gatsby/src/utils/webpack-utils.ts

```diff
@@ -195,12 +195,22 @@ export const createWebpackUtils = (
     },
 
     miniCssExtract: (options = {}) => {
-      return {
-        options,
-        // use MiniCssExtractPlugin only on production builds
-        loader: PRODUCTION
+      let loader: any
+      let hmr = false
+      if (process.env.GATSBY_EXPERIMENTAL_DEV_SSR) {
+        loader = MiniCssExtractPlugin.loader
+        if (stage === `develop`) {
+          hmr = true
+        }
+      } else {
+        loader = PRODUCTION
           ? MiniCssExtractPlugin.loader
-          : require.resolve(`style-loader`),
+          : require.resolve(`style-loader`)
+      }
+
+      return {
+        options: { ...options, hmr },
+        loader,
```

Comment:
Would this whole thing be a bit more readable with something like:
```js
miniCssExtract: (options = {}) => {
  if (PRODUCTION) {
    // production always uses MiniCssExtractPlugin
    return {
     loader: MiniCssExtractPlugin.loader
     options,
    }
  } else if (process.env. GATSBY_EXPERIMENTAL_DEV_SSR) {
    // develop with ssr also uses MiniCssExtractPlugin
   return {
     loader: MiniCssExtractPlugin.loader,
     options: {
       ...options,
       // enable hmr for browser bundle, ssr bundle doesn't need it
       hmr: stage === `develop`, 
     },
    }
  } else {
    // develop without ssr is using style-loader
    return {
      loader: require.resolve(`style-loader`),
      options,
    }
  }
}
```

it's quite hard to understand right now what happens now (took me some time to unpack it, when `loader`/`hmr` was "hoisted" before handling the conditions

Reply:
yes, that does look nicer haha :-D